### ISSUE TITLE
curate the set of available templates

### ISSUE THREAD

**Issue by devoncarew** [google/stagehand#390]
Mentioned in https://github.com/google/stagehand/issues/389:

Another item we've encountered in UX reviews is there are are too many new project templates now (https://github.com/google/stagehand/tree/master/templates) - there's too much choice for the user and not enough clarity on which is the 'right' template. It was always a goal of stagehand to be proscriptive - to have a few high quality templates rather than supporting many different project types and configurations.

We'll likely need to drop a few templates and have clearer guidelines about when new ones are added.

**Comment by devoncarew** [comment]
(we have 8 templates currently - I could see us getting down to 4-5 pretty easily)

**Comment by kwalrath** [comment]
Agreed that there are too many to easily scan. It'd be nice if we could filter templates and/or make it easy to add new ones.

If you're developing a web app, you don't want to see the console app or shelf web server templates. Most people won't need the stagexl app template, but it'd be nice to plug it in if you do want it.

@kevmoo @filiph @chalin we should talk about the templates.

**Comment by devoncarew** [comment]
> If you're developing a web app, you don't want to see the console app or shelf web server templates. 

Even if you're developing a web app, we have 4 web templates to choose from. That's likely too much choice for a new user, w/o the right context to decide between the available options.

**Comment by filiph** [comment]
I see the problem mostly with Angular (2 ways) which I know is about to be addressed. So I vote for having:

* One "AngularDart Web App"
* One "Low-level HTML App"
* One "StageXL Game or App"

Note the changed wording.

**Comment by kwalrath** [comment]
Related issue: #192

**Comment by kwalrath** [comment]
I'm going to take a crack at this as soon as I've updated the docs to not depend on web-angular-simple (https://github.com/dart-lang/site-webdev/issues/591).

**Comment by kwalrath** [comment]
Actually, @chalin is working on this first.

**Comment by kwalrath** [comment]
Do we really need two console samples?

**Comment by devoncarew** [comment]
> Do we really need two console samples?

+1 to dropping one; I haven't looked at those samples for a while, but I suspect generating a sample with a test would be useful guidance for people.

**Comment by kwalrath** [comment]
@filiph, @mit-mit, @anders-sandholm, @kevmoo what do you think of dropping **console-simple** and changing **console-full** to plain old **console**?

**Comment by filiph** [comment]
I do use `console-simple` for doing things like bug repros. But it's true that for "real" projects (even weekend projects) I'd rather go with `console-full` if I can only get one. So yes, LGTM.

**Comment by devoncarew** [comment]
```
  console-full       - A larger command-line application sample.
  console-simple     - A simple command-line application.
  package-simple     - A starting point for Dart libraries or applications.
  server-shelf       - A web server built using the shelf package.
  web-angular        - A web app built using the latest stable version of Angular.
  web-angular-simple - A minimalist example app used in docs.
  web-simple         - An absolute bare-bones web app.
  web-stagexl        - A basic 2D canvas app with StageXL.
```

From looking through the existing templates, it looks like there's a lot of overlap between the console-full and package-simple templates.

**Comment by kwalrath** [comment]
I wouldn't mind removing one more template. And package-simple is one of the less-used packages. (Only server-shelf and web-stagexl had fewer users.)

**Comment by anders-sandholm** [comment]
Consolidation FTW. Just having console SGTM. 

Also fine IMO to just have console and web-angular and then have "others" where we list the rest, if building a hierarchy/tree structure is possible.

**Comment by kwalrath** [comment]
@filiph are you OK with this description for web-angular:

> AngularDart Web App - a web app with material design components. (in WebStorm)
> web-angular - A web app with material design components. (at command line)

(See https://github.com/google/stagehand/pull/400 for context.)

**Comment by filiph** [comment]
The description LGTM!

I'm not sure about removing Stage XL from the templates. It doesn't duplicate anything and provides a very 'gateway drug' option. I know it doesn't have a lot of traffic right now, and probably never will (compared to AngularDart). But it's not like must have only 2 templates in the end.

**Comment by kwalrath** [comment]
Fixed with #403.
